host_modules/debug_info.py

```diff
@@ -0,0 +1,486 @@
+
+  @staticmethod
+  def _run_command(cmd: str, timeout: int = 20):
+    proc = subprocess.Popen(
+        cmd,
+        shell=True,
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        close_fds=True)
+    try:
+      stdout, stderr = proc.communicate(timeout=timeout)
+    except subprocess.TimeoutExpired:
+      proc.kill()
+      return 1, "command timeout", "command timeout"
+    return proc.returncode, stdout, stderr
+
+  @staticmethod
+  def get_board_type() -> str:
+    rc, stdout, err = DebugInfo._run_command(BOARD_TYPE_CMD)
+    board_type = ""
+    if rc != 0:
+      logger.warning("fail to execute command '%s': %s", BOARD_TYPE_CMD, err)
+    else:
+      board_type = stdout.strip()
+    return board_type
+
+  @staticmethod
+  def get_hostname() -> str:
+    cmd = "hostname"
+    rc, stdout, err = DebugInfo._run_command(cmd)
+    hostname = "switch"
+    if rc != 0:
+      logger.warning("fail to execute command '%s': %s", cmd, err)
+    else:
+      hostname = stdout.strip()
+    return hostname
+
+  @staticmethod
+  def _collect_counter_artifacts(directory: str, prefix: str,
+                                 board_type: str) -> None:
+    counter_artifact_dir = os.path.join(
+        directory,
+        datetime.now().strftime(prefix + "counter_%Y%m%d_%H%M%S"))
+    os.makedirs(counter_artifact_dir, exist_ok=True)
+
+    for cmd in COUNTER_CMDS:
+      rc, _, err = DebugInfo._run_command(cmd.format(counter_artifact_dir), timeout=60)
+      if rc != 0:
+        # Continue the artifact collection in case of error.
+        logger.warning("fail to execute command '%s': %s", cmd, err)
+
+  @staticmethod
+  def _sanitize_filename(filename: str) -> str:
+    r"""Sanitize the filenames to remove or replace illegal character.
+
+    Below is a table of characters and their replacements:
+        Char    Replacement
+        "'"     ""
+        ";"     ""
+        " "     "_"
+        "/"     "+"
+        "|"     "\|"
+
+    Args:
+      filename: = string, a file name to sanitize.
+
+    Returns:
+      string: The sanitized filename.
+    """
+    return filename.replace("'", "").replace(";", "").replace(" ", "_").replace(
+        "/", "+").replace("|", r"\|")
+
+  @staticmethod
+  def _send_component_request(component, dir_name, log_level, producer):
+    components = {}
+    for component_id in COMPONENTS:
+      if component == COMPONENT_ALL or component == component_id:
+        # make the artifact dir in the host
+        host_artifact_dir = os.path.join(ARTIFACT_DIR, dir_name, component_id)
+        os.makedirs(host_artifact_dir, exist_ok=True)
+        
+        # send container artifact dir in component request
+        container_artifact_dir = os.path.join(ARTIFACT_DIR_CONTAINER, dir_name,
+                                              component_id)
+        fvs = swsscommon.FieldValuePairs([(LOG_LEVEL_KEY, log_level)])
+        producer.send(component_id, container_artifact_dir, fvs)
```

Comment:
I don't think we have the support for debug artifacts collection in the components, does this work on an actual switch ?